common-lib/src/main/scala/com/gu/mediaservice/lib/ImageIngestOperations.scala

```diff
@@ -20,19 +20,43 @@ class ImageIngestOperations(imageBucket: String, thumbnailBucket: String, config
 
   import ImageIngestOperations.{fileKeyFromId, optimisedPngKeyFromId}
 
-  def storeOriginal(id: String, file: File, mimeType: Option[MimeType], meta: Map[String, String] = Map.empty)
-                   (implicit logMarker: LogMarker): Future[S3Object] =
-    storeImage(imageBucket, fileKeyFromId(id), file, mimeType, meta)
-
-  def storeThumbnail(id: String, file: File, mimeType: Option[MimeType])
-                    (implicit logMarker: LogMarker): Future[S3Object] =
-    storeImage(thumbnailBucket, fileKeyFromId(id), file, mimeType)
-
-  def storeOptimisedPng(id: String, file: File)
+  def store(storableImage: StorableImage)
+           (implicit logMarker: LogMarker): Future[S3Object] = storableImage match {
```

Comment:
Do we need to handle `StorableBrowserViewableImage` too for an exhaustive pattern match? Would making `StorableImage` a sealed trait get the compiler to catch this for us?

Reply:
We never actually _store_ a _Storable_ BrowserViewableImage, it is something we use to generate a thumb and optionally to create an optimised version.  Name change might be appropriate.